compiler/rustc_mir/src/interpret/place.rs

```diff
@@ -176,7 +176,10 @@ impl<Tag> MemPlace<Tag> {
         cx: &impl HasDataLayout,
     ) -> InterpResult<'tcx, Self> {
         Ok(MemPlace {
-            ptr: self.ptr.ptr_offset(offset, cx)?,
+            ptr: match self.ptr.ptr_offset(offset, cx) {
```

Comment:
I'm assuming this change makes any difference on perf simply because we don't properly optimize this in MIR yet on MIR-opt-level =1, right? I would think we already properly optimize this on Mir-opt-level =3. I could be wrong though.